### PR TITLE
Add full support for BibLaTex

### PR DESCRIPTION
**Pull request by rdicosmo**
Here is a first stable proposal to integrate the biblatex styles in acmart based on the work done in https://github.com/rdicosmo/acmart-biblatex/releases/tag/v1.0-rc4 with the help of @k4rtik and testing by @bennn 

It contains:

- the biblatex styles in the stable version developed with great help of @k4rtik and testing by @bennn
- a proposed addition to samples.dtx in the samples directory that produces two examples using biblatex (acmsmall-biblatex and sigconf-biblatex)
- a modified Makefile that handles all the new files
- a proposed addition to acmart.cls that contains a short explanation of the biblatex style files and their use
All this is based upon the latest version of acmart as of https://github.com/borisveytsman/acmart/commit/fd887a614829bb91b4b02e382394fd26d2641736

### CODE REVIEW

**Comment by k4rtik** [comment]
@rdicosmo when I run `make`, I get:
```
[...]
! Package biblatex Error: Style 'acmauthoryear' not found.

See the biblatex package documentation for explanation.
Type  H <return>  for immediate help.
 ...

l.15894 \RequireBibliographyStyle{\blx@bbxfile}

?
```
Do you know what might be going on?

Addendum: exiting that I get this as well:
```
[...]

! Package biblatex Error: Style 'acmnumeric' not found.

See the biblatex package documentation for explanation.
Type  H <return>  for immediate help.
 ...

l.15894 \RequireBibliographyStyle{\blx@bbxfile}

?
```

**Comment by rdicosmo** [comment]
Hi @k4rtik, let me have a look.

**Comment by rdicosmo** [comment]
I cannot reproduce this... make works fine for me on a fresh checkout using gh pr checkout 462
Are you on f8a1475877f01ec8fa7df1f00c0beaf421595c89 ?

**Comment by k4rtik** [comment]
I am, attaching the logs if they are helpful:

[sample-acmsmall-biblatex.log](https://github.com/borisveytsman/acmart/files/8254524/sample-acmsmall-biblatex.log)
[sample-sigconf-biblatex.log](https://github.com/borisveytsman/acmart/files/8254525/sample-sigconf-biblatex.log)

This might be the source of the issue:
```
Package biblatex Info: Trying to load bibliography style 'acmauthoryear'...
Package biblatex Info: ... file 'acmauthoryear.bbx' not found.
```
I see you have included the file, but it's not being recognized for some reason.

**Comment by k4rtik** [comment]
When I moved the `*.*bx` files to `samples`, it seems to build fine, so it's a path problem. No idea of the solution yet.

**Comment by rdicosmo** [comment]
It would be usefule to have the full log of your "make" command.
I tried to follow exactly the same approach used by acmart in the Makefile, so the *.*bx files are copied from the root directory into the samples directory as a consequence of these stanzas

```
BIBLATEXFILES= $(wildcard *.bbx) $(wildcard *.cbx) $(wildcard *.dbx) $(wildcard *.lbx)
SAMPLEBIBLATEXFILES=$(patsubst %,samples/%,$(BIBLATEXFILES))


samples/%: %
        cp $^ samples

samples/%.bbx: %.bbx
samples/%.cbx: %.cbx
samples/%.dbx: %.dbx
samples/%.lbx: %.lbx

samples/%biblatex.pdf: samples/%biblatex.tex $(SAMPLEBIBLATEXFILES)
        cd $(dir $@) && pdflatex-dev $(notdir $<)
        - cd $(dir $@) && biber $(notdir $(basename $<))
        cd $(dir $@) && pdflatex-dev $(notdir $<)
        cd $(dir $@) && pdflatex-dev $(notdir $<)
        while ( grep -q '^LaTeX Warning: Label(s) may have changed' $(basename $<).log) \
          do cd $(dir $@) && pdflatex-dev $(notdir $<); done
```

**Comment by k4rtik** [comment]
I don't see the invocation of the copy of `*.*bx` files in the following output (I reduced the samples to just the `*-biblatex` ones and prefixed `pdflatex` with `texfot` to make the output concise:
```
❯ make
cd samples; texfot pdflatex samples.ins; cd ..
/usr/local/bin/texfot: invoking: pdflatex samples.ins
This is pdfTeX, Version 3.141592653-2.6-1.40.23 (TeX Live 2022/dev) (preloaded format=pdflatex)
No pages of output.
for texfile in samples/*.tex; do \
		pdffile=${texfile%.tex}.pdf; \
		/Library/Developer/CommandLineTools/usr/bin/make $pdffile; \
	done
texfot pdflatex acmart.ins
/usr/local/bin/texfot: invoking: pdflatex acmart.ins
This is pdfTeX, Version 3.141592653-2.6-1.40.23 (TeX Live 2022/dev) (preloaded format=pdflatex)
No pages of output.
cp acmart.cls samples
cp ACM-Reference-Format.bst samples
cd samples/ && texfot pdflatex-dev sample-acmsmall-biblatex.tex
/usr/local/bin/texfot: invoking: pdflatex-dev sample-acmsmall-biblatex.tex
This is pdfTeX, Version 3.141592653-2.6-1.40.23 (TeX Live 2022/dev) (preloaded format=pdflatex-dev)
! Package biblatex Error: Style 'acmauthoryear' not found.

l.15894 \RequireBibliographyStyle{\blx@bbxfile}
! Emergency stop.
 ...
l.15894 \RequireBibliographyStyle{\blx@bbxfile}
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on sample-acmsmall-biblatex.log.
make[1]: *** [samples/sample-acmsmall-biblatex.pdf] Error 1
cd samples/ && texfot pdflatex-dev sample-sigconf-biblatex.tex
/usr/local/bin/texfot: invoking: pdflatex-dev sample-sigconf-biblatex.tex
This is pdfTeX, Version 3.141592653-2.6-1.40.23 (TeX Live 2022/dev) (preloaded format=pdflatex-dev)
! Package biblatex Error: Style 'acmnumeric' not found.

l.15894 \RequireBibliographyStyle{\blx@bbxfile}
! Emergency stop.
 ...
l.15894 \RequireBibliographyStyle{\blx@bbxfile}
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on sample-sigconf-biblatex.log.
make[1]: *** [samples/sample-sigconf-biblatex.pdf] Error 1
make: *** [ALLSAMPLES] Error 2

```

Also, is the `PDF` variable in the `Makefile` deleted intentionally?

File: Makefile [hunk 1]
```diff
@@ -8,11 +8,12 @@
 
 PACKAGE=acmart
 
-
-PDF = $(PACKAGE).pdf acmguide.pdf
```
Comment (k4rtik) [inline Makefile:12]: This line seems unnecessarily deleted.

File: Makefile [hunk 2]
```diff
@@ -8,11 +8,12 @@
...
+BIBLATEXFILES= $(wildcard *.bbx) $(wildcard *.cbx) $(wildcard *.dbx) $(wildcard *.lbx)
+SAMPLEBIBLATEXFILES=$(patsubst %,samples/%,$(BIBLATEXFILES))
 
 all:  ${PDF} ALLSAMPLES
 
+test: $(SAMPLEBIBLATEXFILES)
```
Comment (k4rtik) [inline Makefile]: This target, if intended to be included in this PR, should be marked as phony.

Comment (rdicosmo) [reply Makefile]: May you try `make distclean` and the `make test` to see if the *.*bx files are properly copied into the samples directory?
Here is what should happen:

```
$ make test
cp acmauthoryear.bbx samples
cp acmnumeric.bbx samples
cp software.bbx samples
cp acmauthoryear.cbx samples
cp acmnumeric.cbx samples
cp acmauthoryear.dbx samples
cp acmnumeric.dbx samples
cp software.dbx samples
cp english-software.lbx samples
cp french-software.lbx samples
```

File: Makefile (hunk 1 above)
Comment (rdicosmo) [reply Makefile:12]: Thanks, will fix this.

File: Makefile (hunk 2 above)
Comment (k4rtik) [reply Makefile]: sure, that works.

Is the intention to make `all` target depend on `test`?

Comment (rdicosmo) [reply Makefile]: Maybe the `make` we use have different ideas of the order of precedence of the rules?
I cant see why, but in any case, I have pushed a variant that should work no matter what.
May you try again with `make distclean` and `make` and see if this works for you ?

Comment (k4rtik) [reply Makefile]: > Maybe the `make` we use have different ideas of the order of precedence of the rules?

It's possible, but my understanding is that when invoked without an explicit target, make picks up the first target only, that is, `all` in this case, `test` was never executed in my understanding.

But your recent fix indeed works.

**Comment by k4rtik** [comment]
@rdicosmo thanks for fixing that issue. I see another one, unsure if it's the same for you. The citations aren't rendered correctly in either of the samples:

Screen Shot 2022-03-15 at 1 50 04 PM

This one is supposed to be authoryear.

The following is supposed to be numeric:

Screen Shot 2022-03-15 at 1 50 29 PM

**Comment by rdicosmo** [comment]
> @rdicosmo thanks for fixing that issue. I see another one, unsure if it's the same for you. The citations aren't rendered correctly in either of the samples:
> 
> Screen Shot 2022-03-15 at 1 50 04 PM
> 
> This one is supposed to be authoryear.
> 
> The following is supposed to be numeric:
> 
> Screen Shot 2022-03-15 at 1 50 29 PM

There is definitely an issue with your `make`, and I think I see which one: for some reason, the rule that is applied is the one for the generic %.pdf files instead of the one for the %biblatex.pdf files, so biber is not invoked, and what you see is biblatex printing the label of the references it cannot resolve.

May you try to do the following?

```
rm  samples/sample-acmsmall-biblatex.pdf
make samples/sample-acmsmall-biblatex.pdf
```

and check that biber is properly invoked, instead of bibtex?

**Comment by k4rtik** [comment]
> There is definitely an issue with your `make`

You are right! I was using the default `make` shipped by macOS:
```
❯ which make
/usr/bin/make

❯ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

When I switched to the more recent GNU make shipped by brew, it seems to run `biber` correctly.
```
❯ which make
/opt/homebrew/opt/make/libexec/gnubin/make

❯ make --version
GNU Make 4.3
Built for arm-apple-darwin21.1.0
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Sorry about the trouble. At least, I hope it's worth documenting this issue here for an old version of `make`.

**Comment by borisveytsman** [comment]
gnu make is much superior to the macos default

**Pull request merged**